Tests/Microsoft.AppCenter.Test.Functional/HttpNetworkAdapter.cs

```diff
@@ -28,13 +28,20 @@ public Task<RequestData> MockRequestByLogType(string logType, HttpResponse respo
 
         public Task<RequestData> MockRequest(Func<RequestData, bool> where, HttpResponse response = null, double delayTimeInSeconds = 20)
         {
-            var cancellationToken = new CancellationTokenSource(TimeSpan.FromSeconds(delayTimeInSeconds));
             var expectedData = new ExpectedData
             {
                 Response = response ?? DefaultHttpResponse,
                 Where = where,
-                TaskCompletionSource = new TaskCompletionSource<RequestData>(cancellationToken)
+                TaskCompletionSource = new TaskCompletionSource<RequestData>()
             };
+
+            // Since we can't directly assign token to a Task created with FromResult,
+            // and we can't work with token in SendAsync because it may or may not be called after this method,
+            // we are registering anonymous cancellation token here.
+            // The only weak point can occur if there's a significant delay between actual Task creation time and calling this method,
+            // but since we only use that in tests, that can be ignored.
+            expectedData.cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(delayTimeInSeconds));
+            expectedData.cancellationTokenSource.Token.Register(() => expectedData.TaskCompletionSource.TrySetCanceled());
```

Comment:
Can we move it into `ExpectedData` (+ Dispose)?

Reply:
Resolved in separate [PR's commit](https://github.com/microsoft/appcenter-sdk-dotnet/pull/1312/commits/35970844bebf672221b1e338b41c1afa079976e8)